### PR TITLE
DEPR: Deleted old functions from div-U discretization

### LINKED ISSUE #1065: Delete methods from biot.py/divU
Also check the documentation.

### PR DESCRIPTION
**Pull request by keileg**
## Proposed changes
Deleted two methods in `DivU` discretization class (in `biot.py`). These were tied to the old assembler-style formulation for mixed-dimensional problems and not in use.

Resolves #1065.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [x] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

**Review by IvarStefansson — COMMENTED** [review]
This file's a real mess style wise. But that's not for now. Where do I find the documentation of the discretisation matrices and how to combine them?

**Comment by keileg** [comment]
It is indeed a mess. I added some comments at the top of the `Mpsa` and `Biot` classes.

File: src/porepy/numerics/fv/mpsa.py [hunk 1]
```diff
@@ -24,7 +24,45 @@
...
+    The method can also be used as a discretization method for a mixed-dimensional
+    problem, but it is recommended to do so through the Ad framework, specifically via
+    the wrapper class ``:class:porepy.numerics.ad.discretization.MpsaAd``. Likewise, the
+    Ad framework is the recommended way to use the method for multiphysics problems, on
+    both single and mixed-dimensional grids.
+
+    The discretization is stored as a set of matrices in the dictionary
+    ``data[pp.DISCRETIZATION_MATRICES][self.keyword]``, see ``:meth:discretize`` for
+    details. The most important of these are the ``stress`` matrix, which is the
+    transmissibility matrix, and the ``bound_stress`` matrix, which is the
```
Comment (IvarStefansson) [inline src/porepy/numerics/fv/mpsa.py:44]:
```suggestion
    stiffness matrix, and the ``bound_stress`` matrix, which is the
```

File: src/porepy/numerics/fv/mpsa.py [hunk 2]
```diff
@@ -24,7 +24,45 @@
...
+    discretization of the boundary conditions. Documentation on how to use these
+    matrices to assemble a linear system can be found ``:meth:assemble_matrix_rhs``. In
+    addition, the discretization provides two matrices, ``bound_displacement_cell`` and
+    ``bound_displacement_face`` that together form a reconstruction of the pressure
```
Comment (IvarStefansson) [inline src/porepy/numerics/fv/mpsa.py:48]:
```suggestion
    ``bound_displacement_face`` that together form a reconstruction of the displacement
```

File: src/porepy/numerics/fv/mpsa.py [hunk 3]
```diff
@@ -24,7 +24,45 @@
...
+    trace at boundaries (external and internal). These can be computed by calling
+
+    .. code-block:: Python
+        bc_values = ... # Boundary condition values, can be a combination of Dirichlet
+                        # and Neumann conditions
+
+        d_cell_center = ... # Compute cell center displacement
+        bound_pressure_cell = data[pp.DISCRETIZATION_MATRICES][self.keyword][
```
Comment (IvarStefansson) [inline src/porepy/numerics/fv/mpsa.py:56]:
```suggestion
        bound_displacement_cell = data[pp.DISCRETIZATION_MATRICES][self.keyword][
```

File: src/porepy/numerics/fv/mpsa.py [hunk 4]
```diff
@@ -24,7 +24,45 @@
...
+            self.bound_displacement_cell_matrix_key
+        ]  # Fetch the discretization matrix, see discretize() for details
+        # Do the same for bound_displacement_face
+
+        # Compute the displacment trace p_trace =
```
Comment (IvarStefansson) [inline src/porepy/numerics/fv/mpsa.py:61]:
```suggestion
        # Compute the displacment trace 
        displacement_trace =
```

**Review by IvarStefansson — APPROVED** [review]

**Pull request merged**